### PR TITLE
Carousel Component

### PR DESCRIPTION
**Pull request by oscarwang20**
### Summary 

- adds `carousel.tsx` from shadcn, see docs [here](https://ui.shadcn.com/docs/components/carousel)
  - in addition to the current properties, the carousel component also takes in variables `canClick` and `offset`. 
  - `canClick` determine whether the user can click on items in the carousel to scroll to them using the `scrollTo` method defined in `carousel.tsx`.
  - `offset` determines the index of the first element of the carousel. By default the carousel is zero-indexed with the first element being the one at the very left
- added a `useScreenSize` hook to help with dynamically changing the highlighted element
- added all page routes in `navbar.tsx` and adjusted to underline current page
- implemented `ImageCarousel` component for the Product page

image
image
image

### Notion/Figma Link 

https://www.notion.so/Carousel-Hero-Component-0-5-sprints-ab360524d0c541f4aedcba75b55f29de?pvs=4

### Test Plan 

- Tested by resizing screen and clicking / dragging the component

### Notes 

- If you plan on using the shadcn component, make sure to have at least one more item than however many is currently displayed on the page (partially visible items count) otherwise it would cycle back and forth between the middle two components
- I added the DTI logo as a placeholder until we get the CornellGo logo

### CODE REVIEW

**Comment by henry-li-06** [comment]
So many un-related formatting changes :(

**Comment by oscarwang20** [comment]
> So many un-related formatting changes :(

I had to bump prettier to 3.2.5 bc it was yelling at me about the type imports / exports :(

File: new-dti-website/components/products/imageCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  alt: string;
+  path: string;
+}
+
+const ImageCarousel = (props: { items: carouselItem[] }) => {
+  const plugin = React.useRef(Autoplay({ delay: 1300, stopOnInteraction: false }));
+  const [currentSlide, setCurrentSlide] = React.useState(0);
+  const [carouselApi, setCarouselApi] = React.useState<CarouselApi>();
+  const { width } = useScreenSize();
+  const highlight_index = width < 1024 ? 1 : 3;
```
Comment (cchrischen) [inline new-dti-website/components/products/imageCarousel.tsx:19]: nit: camelCase

**Review by cchrischen — APPROVED** [review]
This looks great! Thanks for updating the Carousel component from shadcn to make it more reusable for the pages. Can you also make the cursor a pointer on hovering over each Carousel item?

**Pull request merged**